packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1133,20 +1133,30 @@ export class ProjectLifecycleManager {
       return metaState
     }
 
-    const configFileTs = this._pathToFile('cypress.config.ts')
-    const configFileJs = this._pathToFile('cypress.config.js')
-
-    if (fs.existsSync(configFileTs)) {
-      metaState.hasValidConfigFile = true
-      this.setConfigFilePath('cypress.config.ts')
-    }
-
-    if (fs.existsSync(configFileJs)) {
-      metaState.hasValidConfigFile = true
-      if (this._configFilePath) {
+    const potentialFiles = [
+      'cypress.config.ts',
+      'cypress.config.mjs',
+      'cypress.config.cjs',
+      'cypress.config.js',
+    ]
```

Comment:
We will need to move this outside and ensure we loop over this array to add the watchers for people using `mjs` here:

https://github.com/cypress-io/cypress/blob/ae884fb9107138686c0e0d899f8303e79e4759fd/packages/data-context/src/data/ProjectLifecycleManager.ts#L610-L614

Nit: Also may need to consider the language conflict error message, right now it hard-codes js and ts, might need to also consider mjs vs ts?

https://github.com/cypress-io/cypress/blob/10.0-release/packages/data-context/src/data/ProjectLifecycleManager.ts#L1352

Reply:
Yeah I was thinking the same about the language conflict error message. I'll update those.